src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -215,9 +217,128 @@ public SAMSequenceRecord addSequenceAlias(final String originalName,
         mSequenceMap.put(altName, originalSeqRecord);
         return originalSeqRecord;
     }
-
+    
+    /** 
+     * called by `parseInterval`. Parse a coordinate string, remove the comma look at the suffix (bp,kb,mp,gb) to scale,
+     * verify that the result is a positive integer
+     * @author Pierre Lindenbaum 
+     * @param numStr the coordinate
+     * @return the coordinate as integer
+     */
+    private static int parseCoordinate(final String numStr)
+        {
+        long factor = 1L;
+        String num = numStr.replace(",","").trim().toLowerCase();
+        if(num.endsWith("bp"))
+            {
+            factor = 1L;
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("kb"))
+            {
+            factor = 1000L;//1E3
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("mb"))
+            {
+            factor = 1000000L;//1E6
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("gb"))
+            {
+            factor = 1000000000L;//1E9
+            num = num.substring(0,num.length()-2).trim();
+            }
+        
```

Comment:
should you throw an exception if there substring doesn't match any of the things you attempted?